src/shared/seccomp-util.c

```diff
@@ -224,6 +225,16 @@ int seccomp_init_for_arch(scmp_filter_ctx *ret, uint32_t arch, uint32_t default_
         if (r < 0)
                 goto finish;
 
+#ifdef SCMP_FLTATR_CTL_LOG
+        if (seccomp_api_get() >= 3) {
```

Comment:
can you add a comment why you check this here as runtime check?

Reply:
It may not be supported by the kernel we're running on, even if we're built against a libseccomp that supports it. I assume the seccomp_attr_set() would fail reasonably gracefully if we skipped the check, but I wasn't sure.

Reply:
i'd prefer if we just ran the actually relevant call, and handle gracefully if that fails, if that is possible. i.e. check for features, not for versions if possible.

(but not sure if libseccomp will generate an error instantly on seccomp_attr_set() or just later when we actually install the filter. If the latter then the version code you have should be ok)

Reply:
OK, at some point I'll see if I can dig up an old kernel to test with :)

Reply:
Looking at the code, libseccomp checks for support immediately at the point where you request it, so this should be OK to drop. SECCOMP_FILTER_FLAG_LOG was introduced in 4.14 apparently, I'm sure I can find something older than that to test with!